### ISSUE TITLE
Use Boto for reference file download in gatk_germline, gatk_preprocessing as well

### ISSUE THREAD

**Issue by fnothaft** [BD2KGenomics/toil-scripts#143]
Otherwise, we have to have some hacky code around, e.g., https://github.com/BD2KGenomics/toil-scripts/blob/master/src/toil_scripts/adam_gatk_pipeline/align_and_call.py#L477.

**Comment by hannes-ucsc** [comment]
In #119, I added support for s3:// URL in *all* copies of the download_url() function this was merged as part of #123.

**Comment by fnothaft** [comment]
Yeah, it looks like that is right. We have some code that downloads either `http(s)://` or `s3://` files (e.g., https://github.com/BD2KGenomics/toil-scripts/blob/master/src/toil_scripts/gatk_germline/germline.py#L96) and we haven't updated https://github.com/BD2KGenomics/toil-scripts/blob/master/src/toil_scripts/adam_uberscript/input_files.py to use `s3://` URLs.

**Comment by jpdna** [comment]
Am I correct that all that needs to be done is change https://github.com/BD2KGenomics/toil-scripts/blob/master/src/toil_scripts/adam_uberscript/input_files.py from `http://` to `s3://` urls?

I just now tested that indeed all the s3:// urls in input_files.py  do work using `download_from_s3_url()` in isolation so I could make a PR now, but I'd like to test running the real ADAM uberscript pipeline first before I make the PR.

 @fnothaft  - are the instructions at  https://github.com/BD2KGenomics/toil-scripts/tree/master/src/toil_scripts/adam_uberscript up to date that I should try to run a small test - or should I wait to test until your #162 related pickle fix gets merged so I can test cleanly end-to-end?

**Comment by hannes-ucsc** [comment]
> Am I correct that ...

yes. But don't call download_from_s3_url. Call download_url which should in turn call the former.

**Comment by jvivian** [comment]
@jpdna — The download_url() function in the common library supports s3 URLs.  Does that solve this issue?
